File: ThreadsApp/ThreadsApp/Utils/PlistUtils.m
//
//  PlistUtils.m
//  ThreadsApp
//
//  Created by Brooma Service on 02/04/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

#import "PlistUtils.h"

@implementation PlistUtils

+ (NSString *) getStringFrom: (NSString*) dictName byKey: (NSString *) key {
    
    NSString* value = nil;
    
    NSDictionary* plistDict = [[NSBundle mainBundle] objectForInfoDictionaryKey: dictName];
    
    if (plistDict && ![plistDict isEqual: [NSNull null]]) {
        value = [plistDict objectForKey: key];
    }
    
    return value;
}

+ (NSString*) getStringFromPlist: (NSString*) key {
    return [[NSBundle mainBundle] objectForInfoDictionaryKey: key];
}

+ (NSString*) getAppVersion {
    return [self getStringFromPlist:@"CFBundleShortVersionString"];
}

@end

File: ThreadsApp/ThreadsAppSwift/Utils/PlistUtils.swift
//
//  PlistUtils.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 14/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation

@objc class PlistUtils : NSObject {
    
    @objc
    class func getStringFrom(dictName: String, byKey key: String) -> String? {
        
        let dict = Bundle.main.infoDictionary?[dictName] as? Dictionary<String, Any>
        return dict?[key] as? String
    }
    
    @objc
    class func getStringFromPlist(byKey key: String) -> String? {
        return Bundle.main.infoDictionary?[key] as? String
    }
    
    @objc
    class func getAppVersion() -> String {
        return getStringFromPlist(byKey: "CFBundleShortVersionString")! // CFBundleShortVersionString must always exist
    }
}
